pkg/validate/validate.go

```diff
@@ -171,12 +172,26 @@ func IP(ip string) error {
 	return nil
 }
 
-// Interface validates if a string is a valid network interface
-func Interface(iface string) error {
-	if _, err := net.InterfaceByName(iface); err != nil {
-		return fmt.Errorf("%s is not a valid network interface: %s", iface, err)
+// LibvirtInterface validates if an interface is a valid network interface on a libvirt host
+func LibvirtInterface(libvirtURI, ifaceName string) error {
+	conn, err := libvirt.NewConnect(libvirtURI)
+	if err != nil {
+		return errors.Wrap(err, "could not connect to libvirt")
 	}
-	return nil
+
+	ifaces, err := conn.ListAllInterfaces(libvirt.CONNECT_LIST_INTERFACES_INACTIVE |
+		libvirt.CONNECT_LIST_INTERFACES_ACTIVE)
+	if err != nil {
+		return errors.Wrap(err, "could not list libvirt interfaces")
+	}
+
+	for _, iface := range ifaces {
+		if iface, err := iface.GetName(); err == nil && iface == ifaceName {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("could not find interface '%s'", ifaceName)
```

Comment:
Do we want to give the list of the interfaces we did find, in case it's just a typo?

Reply:
Please, apply this suggestion

Reply:
`'%s'` -> `%q`